samples/matter/bridge/src/bridged_device_types/humidity_sensor.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "humidity_sensor.h"
+
+namespace
+{
+DESCRIPTOR_CLUSTER_ATTRIBUTES(descriptorAttrs);
+BRIDGED_DEVICE_BASIC_INFORMATION_CLUSTER_ATTRIBUTES(bridgedDeviceBasicAttrs);
+}; // namespace
+
+using namespace ::chip;
+using namespace ::chip::app;
+
+DECLARE_DYNAMIC_ATTRIBUTE_LIST_BEGIN(humiSensorAttrs)
+DECLARE_DYNAMIC_ATTRIBUTE(Clusters::RelativeHumidityMeasurement::Attributes::MeasuredValue::Id, INT16U, 2, 0),
+	DECLARE_DYNAMIC_ATTRIBUTE(Clusters::RelativeHumidityMeasurement::Attributes::MinMeasuredValue::Id, INT16U, 2,
+				  0),
+	DECLARE_DYNAMIC_ATTRIBUTE(Clusters::RelativeHumidityMeasurement::Attributes::MaxMeasuredValue::Id, INT16U, 2,
+				  0),
+	DECLARE_DYNAMIC_ATTRIBUTE(Clusters::RelativeHumidityMeasurement::Attributes::FeatureMap::Id, BITMAP32, 4, 0),
+	DECLARE_DYNAMIC_ATTRIBUTE_LIST_END();
+
+DECLARE_DYNAMIC_CLUSTER_LIST_BEGIN(bridgedHumidityClusters)
+DECLARE_DYNAMIC_CLUSTER(Clusters::RelativeHumidityMeasurement::Id, humiSensorAttrs, nullptr, nullptr),
+	DECLARE_DYNAMIC_CLUSTER(Clusters::Descriptor::Id, descriptorAttrs, nullptr, nullptr),
+	DECLARE_DYNAMIC_CLUSTER(Clusters::BridgedDeviceBasicInformation::Id, bridgedDeviceBasicAttrs, nullptr,
+				nullptr) DECLARE_DYNAMIC_CLUSTER_LIST_END;
+
+DECLARE_DYNAMIC_ENDPOINT(bridgedHumidityEndpoint, bridgedHumidityClusters);
+
+const EmberAfDeviceType gBridgedHumidityDeviceTypes[] = {
+	{ static_cast<chip::DeviceTypeId>(BridgedDevice::DeviceType::HumiditySensor),
+	  BridgedDevice::kDefaultDynamicEndpointVersion },
+	{ static_cast<chip::DeviceTypeId>(BridgedDevice::DeviceType::BridgedNode),
+	  BridgedDevice::kDefaultDynamicEndpointVersion }
+};
+
+const uint8_t gHumidityDataVersionSize = ArraySize(bridgedHumidityClusters);
+
+HumiditySensorDevice::HumiditySensorDevice(const char *nodeLabel) : BridgedDevice(nodeLabel)
+{
+	mDataVersionSize = gHumidityDataVersionSize;
+	mEp = &bridgedHumidityEndpoint;
+	mDeviceTypeList = gBridgedHumidityDeviceTypes;
+	mDeviceTypeListSize = sizeof(gBridgedHumidityDeviceTypes);
+	mDataVersion =
+		reinterpret_cast<DataVersion *>(chip::Platform::MemoryAlloc(sizeof(DataVersion) * mDataVersionSize));
```

Comment:
Do we really need `reinterpret_cast` here? Cast from `void*` should be explicit, right?

Reply:
It is implicit in C, but in C++ `static_cast` is enough.